python_modules/libraries/dagster-dbt/dagster_dbt/dbt_manifest_asset_selection.py

```diff
@@ -45,6 +45,19 @@ class DbtManifestAssetSelection(AssetSelection, arbitrary_types_allowed=True):
     dagster_dbt_translator: DagsterDbtTranslator
     exclude: str
 
+    def __eq__(self, other):
+        if not isinstance(other, DbtManifestAssetSelection):
+            return False
+
+        # Compare metadata only since it uniquely identifies the manifest and the
+        # full manifest dictionary can be large
+        return (
+            self.manifest["metadata"] == other.manifest["metadata"]
```

Comment:
should we make this .get("metadata") just in case? or is it good to error early?

Reply:
We could also fall back to the old slower implementation in the unlikely event that metadata is not present on either...

Reply:
Did that